### ISSUE TITLE
Incorrectly flagged IDE0059

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#68251]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Incorrectly-flagged-IDE0059/10366370)._

---

```csharp
var file = "value1";
try
{
    file = File.Exists("value2") ? "value2" : throw new Exception();
}
catch
{
    Console.WriteLine("value3");
    return;
}
var content = File.ReadAllText(file);
```

The suggestion said IDE0059: Unnecessary assignment of a value to 'file' and the variable is marked grey. The potential fix removes the variable and replace it with `var _ = "value1";`, which is incorrect.

Even tho the value1 is never used, the existence of the variable file is necessary.

---
#### Yonghan Ruan on 5/17/2023, 06:25 PM: 

(private comment, text removed)
#### Feedback Bot on 5/18/2023, 00:48 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by Youssef1313** [comment]
The diagnostic looks correct to me. But the codefix should be replacing `var file = "value1";` with `string file;`

**Comment by MisinformedDNA** [comment]
Yep, `file` will always be equal to `value2` on the last line.

I pasted the code into VS and the analyzer gave me "Remove redundant assignment" and it became `string? file;`, which is correct.

@sharwell Propose close?

**Comment by CyrusNajmabadi** [comment]
Agreed.  Going ot keep open though as i think it would be good to have a test for this case added to show that it's intentional, and to prevent it from breaking.
